src/CaptainsLog/campaign/intel/RuinsIntel.java

```diff
@@ -1,225 +1,6 @@
-    }
-
-    @Override
-    public void createSmallDescription(TooltipMakerAPI info, float width, float height) {
-        float opad = 10f;
-
-        TooltipMakerAPI text = info.beginImageWithText(ruinsType.getIcon(), 48);
-        text.addPara(Misc.getTokenReplaced(ruinsType.getDesc(), marketToken), Misc.getGrayColor(), opad);
-        info.addImageWithText(opad);
-
-        info.addPara("Located in the " + marketToken.getStarSystem().getNameWithLowercaseType() + ".", opad,
-                Misc.getPositiveHighlightColor(), marketToken.getStarSystem().getBaseName());
-
-        if (!isEnding()) {
-            addGenericButton(info, width, ignoreButtonCaption, IGNORE);
-        }
-//        addGenericButton(info, width, deleteButtonCaption, IGNORE_ALL);
-    }
-
-    @Override
-    public Set<String> getIntelTags(SectorMapAPI map) {
-        Set<String> tags = super.getIntelTags(map);
-        tags.add(INTEL_RUINS);
-        return tags;
-    }
-
-    @Override
-    public SectorEntityToken getMapLocation(SectorMapAPI map) {
-        if (isEnding()) {
-            return null;
-        } else {
-            return marketToken.getMarket().getStarSystem().getHyperspaceAnchor();
-        }
-    }
-
-    @Override
-    public String getCommMessageSound() {
-        return "ui_discovered_entity";
-    }
-
-    public SectorEntityToken getEntity() {
-        return marketToken;
-    }
-
-    @Override
-    public IntelSortTier getSortTier() {
-        if (isEnding()) {
-            return IntelSortTier.TIER_COMPLETED;
-        }
-        return IntelSortTier.TIER_4;
-    }
-
-    private int getRuinsTier() {
-        switch (ruinsType.getId()) {
-            case (Conditions.RUINS_SCATTERED):
-                return 1;
-            case (Conditions.RUINS_WIDESPREAD):
-                return 2;
-            case (Conditions.RUINS_EXTENSIVE):
-                return 3;
-            case (Conditions.RUINS_VAST):
-                return 4;
-            default:
-                return 0;
-        }
-    }
-
-    private float getFraction() {
-        return getRuinsTier() / 4f;
-    }
-
-    @Override
-    public String getSortString() {
-        return "Ruins " + getRuinsTier();
-    }
-
-    @Override
-    public void buttonPressConfirmed(Object buttonId, IntelUIAPI ui) {
-        if (buttonId == IGNORE) {
-            MemoryAPI mem = this.marketToken.getMemory();
-            mem.set(IGNORE_RUINS_MEM_FLAG, true);
-            endImmediately();
-
-            if (ui != null) {
-                ui.recreateIntelUI();
-            }
-        }
-    }
-
-    @Override
-    public boolean doesButtonHaveConfirmDialog(Object buttonId) {
-        return true;
-    }
-
-    @Override
-    public void createConfirmationPrompt(Object buttonId, TooltipMakerAPI prompt) {
-        if (buttonId == IGNORE) {
-            prompt.addPara(ignoreConfirmMessage, 0f); // TODO
-        }
-    }
-}
+@Deprecated
+public class RuinsIntel extends BaseIntelPlugin {}
```

Comment:
This class is only needed for legacy save-games. Since the move to RuinsIntelv2 happened back in 0.9.1a, it could even be fully removed (as save-games are not cross-version compatible).

Reply:
Even if you'd want to keep it, it's only needed for deserializer, and all instances are removed on game load - thus empty implementation is enough for cleanup to happen.